### DISCUSSION TITLE
Provide GitHub-hosted runners in other regions

### DISCUSSION THREAD

**Discussion by aramfe** [community/community#11727] [Actions]
Right now GitHub-hosted runners are located in the US only. 

This leads to massive latency issues for HTTP requests against servers in non-US regions like Europe. 

In our case, especially our end-to-end tests, which run from GitHub-hosted runners against our deployments in an ECS cluster in Europe, are heavily impacted by these location-induced latency problems: The tests not only got very flaky, but are also extremely slow in their execution time. While the tests run locally in like two minutes, they take like eight minutes on GitHub Actions. Even with sharding and parallelizing them, they are still much slower than expected.

Before we switched from AWS CodePipeline to GitHub Actions, we've never encountered these kind of problems, as CodeBuild runners were always hosted in the same region, the pipeline was defined in. We actually expected this to be the case here as well. One of the main selling points for us, was the performance of the GitHub runners themselves: They are very fast and also very stable. Having self-hosted runners on the other hand, would lead to a lot of maintenance overhead. Well. Not having runners in the same region our product is deployed in, basically forces us to setup self-hosted runners -> Which is obviously diminishing this selling point completely. 

I hope for a solution regarding this, as I think that this is heavily needed for non-US users as well.

**Comment by keithlai124** [comment]
Yes, I definitely agree that we need this feature ASAP. 

Currently using GitHub-hosted runners in Australia and would desperately want some runners in the same region that our product is deployed in.

**Comment by ethomson** [comment]
Thanks for the feedback - I can understand your desire to select a region for your runners.  We have this on our backlog, but we don't have any immediate plans for this.

**Reply by ashikkumar23** [reply]
Hi @ethomson, 

An ETA for this feature would be highly helpful. Thank you.

**Comment by guillaume-gagnaire** [comment]
Any ETA about this feature ? It's a must-have ! :)

**Comment by mike-kaminski** [comment]
@aramfe fwiw, running self-hosted runners on ECS/Fargate as a service for the past 6months or so has been more or less completely maintenance free. The runner agent self-updates by default and the container's init script auto-registers on startup, which allows me to scale the runner instances up and down (I use an EventBridge scheduled task to scale up to 10 runners during business hours and back down to 1 off hours). I also chucked an API call in to deregister any offline runners into the init script, automating the cleanup within the same process. You can't run any Actions that rely on Docker, but for E2E and browser tests (and pretty much most things), that's not been a problem.

**Comment by zyv** [comment]
This has already been requested several times in the past:

* https://github.com/orgs/community/discussions/24969
* https://github.com/orgs/community/discussions/25994

We also absolutely need the ability to pin the region to be able to utilise GitHub-hosted runners for our data processing workloads due to GDPR compliance requirements.

With the new powerful runners that have just been released, it's essential that you implement this so that many users can make full use of it for other purposes than CI as such.

**Comment by alexlaverty** [comment]
Any updates? Would also like the ability to have github workflow actions running from australian servers.

**Reply by jspencer-analytics** [reply]
Hi there, have you found a work around? Im having the same issue where i need code to run in Australian region

**Comment by gaurav517** [comment]
Any updates on this please? (request for Australian region). Thanks.

**Reply by jspencer-analytics** [reply]
Hi there, have you found a work around? Im having the same issue where i need code to run in Australian region

**Comment by soulmachine** [comment]
Any updates on this? Some websites ban US IP addresses and then all RESTful requests return HTTP 451 error, need this feature ASAP

**Reply by oliver-zehentleitner** [reply]
Same here.... US servers of github actions are blocked in our tests...

**Reply by ornakash** [reply]
Same here!

**Comment by Playback80** [comment]
> ตอนนี้นักวิ่งที่โฮสต์โดย GitHub อยู่ในสหรัฐอเมริกาเท่านั้น
> 
> สิ่งนี้นำไปสู่ปัญหาเวลาแฝงขนาดใหญ่สำหรับคำขอ HTTP กับเซิร์ฟเวอร์ในภูมิภาคที่ไม่ใช่ของสหรัฐอเมริกา เช่น ยุโรป
> 
> ในกรณีของเรา โดยเฉพาะอย่างยิ่งการทดสอบแบบ end-to-end ซึ่งรันจากรันเนอร์ที่โฮสต์บน GitHub เทียบกับการปรับใช้ของเราในคลัสเตอร์ ECS ในยุโรป ได้รับผลกระทบอย่างมากจากปัญหาความหน่วงที่เกิดจากตำแหน่งที่ตั้ง: การทดสอบไม่เพียงแต่มีความไม่สม่ำเสมอเท่านั้น แต่ ยังช้ามากในเวลาดำเนินการ ในขณะที่การทดสอบดำเนินการในเครื่องภายในสองนาที พวกเขาใช้เวลาแปดนาทีใน GitHub Actions แม้จะมีการแตกย่อยและขนานกัน แต่ก็ยังช้ากว่าที่คาดไว้มาก
> 
> ก่อนที่เราจะเปลี่ยนจาก AWS CodePipeline เป็น GitHub Actions เราไม่เคยพบปัญหาประเภทนี้ เนื่องจากตัวเรียกใช้ CodeBuild มักจะโฮสต์ในภูมิภาคเดียวกัน ไปป์ไลน์ถูกกำหนดไว้แล้ว อันที่จริง เราคาดว่าจะเป็นเช่นนั้นที่นี่เช่นกัน หนึ่งในจุดขายหลักสำหรับเราคือประสิทธิภาพของตัวรัน GitHub เอง: พวกมันเร็วมากและเสถียรมาก ในทางกลับกัน การมีนักวิ่งที่โฮสต์เองจะนำไปสู่ค่าใช้จ่ายในการบำรุงรักษาจำนวนมาก ดี. การไม่มีรันเนอร์ในภูมิภาคเดียวกันที่ผลิตภัณฑ์ของเราถูกนำไปใช้งาน โดยทั่วไปบังคับให้เราต้องตั้งค่ารันเนอร์ที่โฮสต์เอง -> ซึ่งเห็นได้ชัดว่าลดจุดขายนี้ลงอย่างสิ้นเชิง
> 
> ฉันหวังว่าจะมีวิธีแก้ปัญหาเกี่ยวกับเรื่องนี้ เนื่องจากฉันคิดว่านี่เป็นสิ่งจำเป็นอย่างยิ่งสำหรับผู้ใช้ที่ไม่ได้อยู่ในสหรัฐอเมริกาเช่นกัน

**Comment by lnxslck** [comment]
Any update on this?
I have a few github actions that stopped working since the API they were using is now banning US Ips

**Comment by the-patience-tester** [comment]
hi, Any update on this?

**Comment by alexhales555** [comment]
Yes, I definitely agree that we need this feature as soon as possible. Currently, we are using GitHub-hosted runners in Australia. However, we urgently need runners located in the same region where our product, [WhatsApp Plus](https://apkdon.net/whatsapp-plus/), is deployed.

**Comment by adamaiken89** [comment]
It will be helpful because we can run some regular jobs on the respective regions. Such performance gain is massively and we do not need to configure the dedicated runners anymore.

**Comment by lkfortuna** [comment]
Hi there, we do not have plans in the next 3 months to enable regional actions runners. Thank you all for the feedback ad follow up. I will follow up here in 3 months time with an update in this space.

**Reply by qt-pixels** [reply]
I feel like we should be reimbursed for all the failed GHAs, due to e2e tests exceeding fair defaults (5s). We're sometimes re-running tests, which end up increasing our Lead Time for Change metrics because of this.

**Reply by Livan-pro** [reply]
It seems like 4 months has passed. Any updates?

**Reply by mohitchaniyal** [reply]
Any updates?

**Reply by YorhaJ** [reply]
any updates on this?

**Reply by lkfortuna** [reply]
Hi there- an update on this is: GitHub released Enterprise cloud with data residency in October: https://github.blog/changelog/2024-10-29-github-enterprise-cloud-data-residency-in-the-eu-is-generally-available/

This is our answer to the data residency and regional runners question. This data residency option will be expanding to other regions over the course of the next year. As each region is released, there are region specific runners available in that region.

**Reply by the-patience-tester** [reply]
@lkfortuna That only refers to data location not hosted runner location, that means to use githubs hosted runners for anything like automated testing we would have to open up geo-locked software to be accessed in the US as that's currently where the runners reside. So it doesn't actually address the issue.

**Reply by lkfortuna** [reply]
the runners are also hosted in the region that the data residency is in.

**Reply by gaurav517** [reply]
Is it possible to use Github runner in a region (e.g. Australia) without using data residency for that region? 
For us, we can't use data residency because our idP (Google) is not supported for EMU (Enterprise Managed Users) feature and using EMU is mandatory for using data residency.

**Comment by pepitoenpeligro** [comment]
Hi team, any updates on this? This feature could be a big deal for GitHub Organizations located outside U.S.
Thank you so much!

**Reply by lkfortuna** [reply]
See above https://github.com/orgs/community/discussions/11727#discussioncomment-11719412

**Comment by netomarchiori** [comment]
It would be very beneficial now that there's a cost for using self-hosted runners.

I use self-hosted because some hosting companies restrict ssh access to their servers from IPs not in Brazil.

**Reply by fuomag9** [reply]
> It would be very beneficial now that there's a cost for using self-hosted runners.
> 
> I use self-hosted because some hosting companies restrict ssh access to their servers from IPs not in Brazil.

FYI they backtracked on this already, but not canceled it, only delayed for now
